### PR TITLE
feat: return array of data and error for revalidate and fetchNext

### LINKED ISSUE #133: accessor.revalidate should return the whole fetch result
Currently it only return the data or `null`. It should also return error when fetching fails. It returns `null` only when the fetching is interrupted by the other fetching.

### PR DESCRIPTION
**Pull request by jason89521**
## Proposed change

close #133

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Testing
- [ ] Refactor
- [ ] Chore (tool changes, configuration changes, and changes to things that do not actually go into production at all)

### CODE REVIEW

File: src/lib/model/InfiniteAccessor.ts [hunk 1]
```diff
@@ -182,11 +181,11 @@ export class InfiniteAccessor<S, Arg = any, Data = any, E = unknown> extends Acc
         }
         this.currentTask = 'idle';
         this.onFetchingFinish();
-        return data;
+        return error ? ([null, error] as const) : ([data, null] as const);
```
Comment (Fonger) [inline src/lib/model/InfiniteAccessor.ts:184]: It's more common that the error is placed in the first argument, isn't it?

Comment (jason89521) [reply src/lib/model/InfiniteAccessor.ts:184]: In what situations does it make more sense to put error before data? Could you provide an example?

Comment (jason89521) [reply src/lib/model/InfiniteAccessor.ts]: I put error before data and make a more typesafe definition in [76c279c](https://github.com/jason89521/daxus/pull/137/commits/76c279ce73cc836898ac4b55884e73c0e32e1141)

Comment (Fonger) [reply src/lib/model/InfiniteAccessor.ts]: Like built-in node.js api like `fs.readFile`. If you don't use fs/promise, you can receive a callback which is in form of `function(err, data) {...}`. As for third-party packages, `mongoose` follows the same pattern.

For other languages like golang lack exception-based error handling (try/catch), they usually `return nil, data` or  `return err, nil` when an error occurs.

**Pull request merged**